scripts/lspArtifact.ts

```diff
@@ -0,0 +1,174 @@
+        const manifestUrl = 'https://aws-toolkit-language-servers.amazonaws.com/qAgenticChatServer/0/manifest.json'
+
+        https
+            .get(manifestUrl, (res) => {
+                let data = ''
+
+                res.on('data', (chunk) => {
+                    data += chunk
+                })
+
+                res.on('end', async () => {
+                    try {
+                        const manifest: Manifest = JSON.parse(data)
+
+                        const latestVersion = manifest.versions
+                            .filter((v) => !v.isDelisted)
+                            .sort((a, b) => b.serverVersion.localeCompare(a.serverVersion))[0]
+
+                        if (!latestVersion) {
+                            throw new Error('No valid version found in manifest')
+                        }
+
+                        const darwinArm64Target = latestVersion.targets.find(
+                            (t) => t.platform === 'darwin' && t.arch === 'arm64'
+                        )
+
+                        if (!darwinArm64Target) {
+                            throw new Error('No darwin arm64 target found')
+                        }
+
+                        for (const content of darwinArm64Target.contents) {
+                            const fileName = content.filename
+                            const fileUrl = content.url
+                            const expectedHash = content.hashes[0]
+                            const tempFilePath = path.join(tempDir, fileName)
+                            const fileFolderName = content.filename.replace('.zip', '')
+
+                            console.log(`Downloading ${fileName} from ${fileUrl} ...`)
+
+                            await new Promise((downloadResolve, downloadReject) => {
+                                https
+                                    .get(fileUrl, (fileRes) => {
+                                        const fileStream = fs.createWriteStream(tempFilePath)
+                                        fileRes.pipe(fileStream)
+
+                                        fileStream.on('finish', () => {
+                                            fileStream.close()
+                                            downloadResolve(void 0)
+                                        })
+
+                                        fileStream.on('error', (err) => {
+                                            fs.unlink(tempFilePath, () => {})
+                                            downloadReject(err)
+                                        })
+                                    })
+                                    .on('error', (err) => {
+                                        fs.unlink(tempFilePath, () => {})
+                                        downloadReject(err)
+                                    })
+                            })
+
+                            console.log(`Verifying hash for ${fileName}...`)
```

Comment:
should switch to logger?

Reply:
this is inside the package script so it should do console.log